src/test/java/seedu/address/logic/commands/FilterCommandTest.java

```diff
@@ -57,32 +72,70 @@ public void equals() {
+        Set<NameContainsKeywordsPredicate> namePredicates = prepareNamePredicate("Kurz,Elle,Kunz");
+        Set<TagMatchesQueryPredicate> tagPredicates = prepareTagPredicate("owesMoney, friends");
+
+        FilterCommandPredicate predicate1 = new FilterCommandPredicate(namePredicates, null);
+        FilterCommand command1 = new FilterCommand(predicate1);
+        String expectedMessage1 = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 3);
+        namePredicates.forEach(pred -> expectedModel.addNewFilterToFilteredPersonList(pred));
+        assertCommandSuccess(command1, model, expectedMessage1, expectedModel);
         assertEquals(Arrays.asList(CARL, ELLE, FIONA), model.getFilteredPersonList());
+
+        expectedModel.clearFiltersInFilteredPersonList();
+        model.clearFiltersInFilteredPersonList();
+
+        FilterCommandPredicate predicate2 = new FilterCommandPredicate(null, tagPredicates);
+        FilterCommand command2 = new FilterCommand(predicate2);
+        String expectedMessage2 = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 3);
+        tagPredicates.forEach(pred -> expectedModel.addNewFilterToFilteredPersonList(pred));
+        assertCommandSuccess(command2, model, expectedMessage2, expectedModel);
+        assertEquals(Arrays.asList(ALICE, BENSON, DANIEL), model.getFilteredPersonList());
+
+        expectedModel.clearFiltersInFilteredPersonList();
+        model.clearFiltersInFilteredPersonList();
+
+        FilterCommandPredicate predicate3 = new FilterCommandPredicate(namePredicates, tagPredicates);
+        FilterCommand command3 = new FilterCommand(predicate3);
+        String expectedMessage3 = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 6);
+        namePredicates.forEach(pred -> expectedModel.addNewFilterToFilteredPersonList(pred));
+        tagPredicates.forEach(pred -> expectedModel.addNewFilterToFilteredPersonList(pred));
+        assertCommandSuccess(command3, model, expectedMessage3, expectedModel);
+        assertEquals(Arrays.asList(ALICE, BENSON, CARL, DANIEL, ELLE, FIONA), model.getFilteredPersonList());
+    }
+
+    /**
+     * Parses {@code userInput} into a set of {@code NameContainsKeywordsPredicate}.
+     */
+    private Set<NameContainsKeywordsPredicate> prepareNamePredicate(String userInput) {
+        return Stream.of(userInput.split(COMMA)).map(name -> new NameContainsKeywordsPredicate(name))
+                .collect(Collectors.toSet());
     }
 
     /**
-     * Parses {@code userInput} into a {@code NameContainsKeywordsPredicate}.
+     * Parses {@code userInput} into a set of {@code TagMatchesQueryPredicate}.
      */
-    private NameContainsKeywordsPredicate preparePredicate(String userInput) {
-        return new NameContainsKeywordsPredicate(Arrays.asList(userInput.split("\\s+")));
+    private Set<TagMatchesQueryPredicate> prepareTagPredicate(String userInput) {
+        return Stream.of(userInput.split(COMMA)).map(input -> new Tag(input))
```

Comment:
Minor style gripe: maybe it would be cleaner to put each functional method in a separate line:

```
Stream.of(...)
    .map(...)
    .map(...)
    .collect(...)
```